### PR TITLE
Fix AccessMode name in NSX request body

### PR DESCRIPTION
**Pull request by wenqiq**
Fix AccessMode name in NSX request body when creating subnet and fix subnet e2e test

### CODE REVIEW

**Review by zhengxiexie — APPROVED** [review]
LGTM

File: pkg/apis/vpc/v1alpha1/vpcnetworkconfiguration_types.go [hunk 1]
```diff
@@ -9,9 +9,10 @@ import (
 )
 
 const (
-	AccessModePublic  string = "Public"
-	AccessModePrivate string = "Private"
-	AccessModeProject string = "PrivateTGW"
+	AccessModePublic       string = "Public"
+	AccessModePrivate      string = "Private"
+	AccessModeProject      string = "PrivateTGW"
+	AccessModeProjectInNSX string = "Private_TGW"
```
Comment (lxiaopei) [inline pkg/apis/vpc/v1alpha1/vpcnetworkconfiguration_types.go:15]: we should remove podSubnetAccessMode in VPCNetworkConfiguration CR.

File: pkg/nsx/services/subnet/builder.go [hunk 2]
```diff
@@ -44,6 +44,13 @@ func (service *SubnetService) buildSubnetSetName(subnetset *v1alpha1.SubnetSet,
 	return util.GenerateTruncName(common.MaxSubnetNameLength, resName, "", index, "", "")
 }
 
+func convertAccessMode(accessMode string) string {
```
Comment (lxiaopei) [inline pkg/nsx/services/subnet/builder.go:47]: we do not need this func. NSX VPC subnet has the below access mode, in Subnet/SubnetSet CR, we should keep the same access mode.
access_mode:
      type: string
      enum:
        - Private
        - Public
        - Isolated
        - Private_TGW

File: pkg/apis/vpc/v1alpha1/vpcnetworkconfiguration_types.go [hunk 3]
```diff
@@ -9,9 +9,10 @@ import (
 )
 
 const (
-	AccessModePublic  string = "Public"
-	AccessModePrivate string = "Private"
-	AccessModeProject string = "PrivateTGW"
+	AccessModePublic       string = "Public"
```
Comment (lxiaopei) [inline pkg/apis/vpc/v1alpha1/vpcnetworkconfiguration_types.go:12]: AccessMode should not be added here, it should be in subnet_type.go

File: pkg/apis/vpc/v1alpha1/vpcnetworkconfiguration_types.go (hunk 1 above)
Comment (lxiaopei) [inline pkg/apis/vpc/v1alpha1/vpcnetworkconfiguration_types.go:15]: you do not need define access mode here, you need to define them in subnet_type.go

Comment (wenqiq) [reply pkg/apis/vpc/v1alpha1/vpcnetworkconfiguration_types.go:15]: https://github.com/vmware-tanzu/nsx-operator/issues/680
will update in another PR.

File: pkg/apis/vpc/v1alpha1/subnet_types.go [hunk 4]
```diff
@@ -9,6 +9,13 @@ import (
 
 type AccessMode string
 
+const (
+	AccessModePublic       string = "Public"
+	AccessModePrivate      string = "Private"
+	AccessModeProject      string = "PrivateTGW"
+	AccessModeProjectInNSX string = "Private_TGW"
```
Comment (dantingl) [inline pkg/apis/vpc/v1alpha1/subnet_types.go:16]: I don't think we should define the NSX type in CRD API file

**Review by heypnus — APPROVED** [review]

**Review by zhengxiexie — APPROVED** [review]

File: pkg/apis/vpc/v1alpha1/vpcnetworkconfiguration_types.go [hunk 5]
```diff
@@ -8,12 +8,6 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-const (
-	AccessModePublic  string = "Public"
-	AccessModePrivate string = "Private"
-	AccessModeProject string = "PrivateTGW"
-)
-
```
Comment (dantingl) [inline pkg/apis/vpc/v1alpha1/vpcnetworkconfiguration_types.go:16]: Revert these changes?

File: test/e2e/manifest/testSubnet/subnetport_1.yaml [hunk 6]
```diff
@@ -1,7 +1,7 @@
 apiVersion: crd.nsx.vmware.com/v1alpha1
 kind: SubnetPort
 metadata:
-  name: port-1
+  name: port-e2e-test-1
```
Comment (dantingl) [inline test/e2e/manifest/testSubnet/subnetport_1.yaml:4]: What's the reason to change the name?
